frontend/src/components/pages/agents/details/a2a/chat/hooks/streaming-types.ts

```diff
@@ -152,6 +152,10 @@ export type StreamingState = {
    * Latest task state from status updates
    */
   capturedTaskState: ChatMessage['taskState'] | undefined;
+  /**
+   * Previous task state for detecting state transitions
+   */
+  previousTaskState: string | undefined;
```

Comment:
[nit] task state could technically be `'submitted' | 'working' | 'input-required' | 'completed' | 'canceled' | 'failed' | 'rejected' | 'auth-required' | 'unknown'`, can be taken from `TaskStateProps['state']`